registration/signals.py

```diff
@@ -54,3 +58,38 @@ def password_reset_token_created(
         subject=settings.RESET_PASSWORD_EMAIL_SUBJECT,
         recipient_list=(reset_password_token.user.email,)
     )
+
+
+@receiver(appserver_spawned)
+def send_acknowledgement_email_on_instance_provisioning(sender, **kwargs):
+    """
+    Send acknowledgement email for successful instance provisioning, except
+    for periodic builds.
+    """
+    instance = kwargs['instance']
+    appserver = kwargs['appserver']
+    deployment_id = kwargs['deployment_id']
+
+    if deployment_id is None or appserver is None:
+        return
+
+    deployment = Deployment.objects.get(pk=deployment_id)
+
+    # Send emails for only user triggered emails
+    if deployment.type != DeploymentType.user.name:
+        return
+
+    application = instance.betatestapplication_set.first()  # There should only be one
+    is_appserver_healthy = appserver.status.is_healthy_state
+
+    # Send email only for registered client application where application status is
+    # accepted. Applications with pending status will go through the approval signal
+    # and send welcome email.
+    if not application or application.status != BetaTestApplication.ACCEPTED:
```

Comment:
Can an app server spawn if the BetaTestApplication status is pending?

Reply:
The first deployment after the registration happens with `BetaTestApplication.PENDING` status (Welcome email sent on success).
This check serves as a check to determine that current deployment is a config update deployment.
In theory, appserver spawns are independent of `BetaTestApplication` status.